### PR TITLE
output_filename virtualbox option.

### LINKED ISSUE #9145: virtualbox-iso: allow to specify a name for artifact file different than vm_name
#### Feature Description

When you build a virtual machine, you may want to have a name for the artifact different than name of the machine. Currently (Packer 1.5.6), it's not possible. The `VBoxManager export` command used by Packer use the `vm_name` setting as filename of artifact.

#### Use Case(s)

I'm building a virtual machine using `virtualbox-iso` builder and I want to add to my artifact filename, the version of the box.

### PR DESCRIPTION
**Pull request by SwampDragons**
Allow users to set the output filename instead of forcing the filename to be the same as the vm_name. 

Closes #9145

### CODE REVIEW

File: builder/virtualbox/common/step_export_test.go [hunk 1]
```diff
@@ -43,3 +43,71 @@ func TestStepExport(t *testing.T) {
...
+			Expected: "output-dir/output-filename.ova",
+			Reason:   "output_filename should not be vmName if set.",
+		},
+		{
+			Step: &StepExport{
+				Format:         "ovf",
+				OutputDir:      "output-dir",
+				OutputFilename: "",
+			},
+			Expected: "output-dir/foo.ovf",
```
Comment (sylviamoss) [inline builder/virtualbox/common/step_export_test.go:69]: I guess you need to use the `filepath.Join(...)` here to use the right OS path separator.

**Comment by nqb** [comment]
We will need to update docs accordingly.

**Comment by sylviamoss** [comment]
@nqb The docs will be generated from the comment above the variable

**Comment by SwampDragons** [comment]
oops forgot to commit the generated files 😅

**Comment by orcutt989** [comment]
Can we get this in the vsphere post-processor?

**Comment by sylviamoss** [comment]
@orcutt989 could open an issue for this, please? it will be easier to track your request.

**Comment by nqb** [comment]
@SwampDragons, changelog entry is missing.

**Comment by sylviamoss** [comment]
@nqb changelog is now updated

**Pull request merged**